### ISSUE TITLE
Feature request: Allow selection & copy/paste in preview mode

### ISSUE THREAD

**Issue by andreasmjg** [etesync/etesync-notes#63]
Allow selection & copy/paste in preview mode

**Comment by tasn** [comment]
I wonder if it's even possible given how the markdown rendering works. We'll have to take a look at it. Thanks!

**Comment by zecakeh** [comment]
I'm guessing this requires overriding the render rules of inline elements in `react-native-markdown-display`. It would be easier to fix that upstream.

**Comment by tasn** [comment]
Yeah, you are probably right, though I wonder how it works with the multiple text spans. I think we will probably need a wrapping text object so that it's all part of the same text view (and thus can be selected).

**Comment by zecakeh** [comment]
The problem is you can't have a `View` inside a `Text`, if I remember correctly there's a bug on Android.
I think it would work, but you wil probable lose the blocks and just have inline text.

**Comment by tasn** [comment]
Yeah, you can't. I don't think it's a bug, just a limitation. I guess views are used for some of the things, so we can't wrap with a text object.
This means that it's potentially not possible to select across text spans (e.g. lines) in the markdown preview, but it should be possible to select single words and lines I think, so at least we should have that. Just sharing what I remember, but I could be wrong.

**Comment by zecakeh** [comment]
I've been looking at making a PR to add that feature at `react-native-markdown-display`. As you thought, you can only select by line or paragraph. As soon as you change block, you can't select anymore.

Should we have the preview as HTML in a WebView to not be limited by that ?

**Comment by tasn** [comment]
I'm not happy about moving to using a webview because of memory consumption and performance, though I guess we will be going that direction anyway because of the rich text editor. I really don't know what to say. :\ 
Can you ping me again about this tomorrow and we'll have another think?

**Comment by zecakeh** [comment]
Another solution might be to have everything embedded in a single Text component, that means no views. That's doable, if we change the styles a bit (except for tables and images of course). That would mean a completely different renderer. And we wouldn't be that far from a native WYSIWYG editor 😛
